llvm/include/llvm/Option/OptStrCmp.h

```diff
@@ -0,0 +1,38 @@
+//===- OptStrCmp.h - Option String Comparison -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_OPTION_OPTSTRCMP_H
+#define LLVM_OPTION_OPTSTRCMP_H
+
+#include "llvm/ADT/StringRef.h"
+
+namespace llvm::opt {
+
+// Comparison function for Option strings (option names & prefixes).
+// The ordering is *almost* case-insensitive lexicographic, with an exception.
+// '\0' comes at the end of the alphabet instead of the beginning (thus options
+// precede any other options which prefix them). Additionally, if two options
+// are identical ignoring case, they are ordered according to case sensitive
+// ordering if `FallbackCaseSensitive` is true.
+static int StrCmpOptionName(StringRef A, StringRef B,
+                            bool FallbackCaseSensitive = true) {
+  size_t MinSize = std::min(A.size(), B.size());
+  if (int Res = A.substr(0, MinSize).compare_insensitive(B.substr(0, MinSize)))
+    return Res;
+
+  // If they are identical ignoring case, use case sensitive ordering.
+  if (A.size() == B.size())
+    return FallbackCaseSensitive ? A.compare(B) : 0;
+
+  return (A.size() == MinSize) ? 1 /* A is a prefix of B. */
+                               : -1 /* B is a prefix of A */;
+}
+
+} // namespace llvm::opt
+
+#endif // LLVM_OPTION_OPTSTRCMP_H
```

Comment:
Missing newline at the end of the file. I'm surprised clang-format doesn't complain here.
```suggestion
#endif // LLVM_OPTION_OPTSTRCMP_H

```

Reply:
Done.